tests/python_eol/Dockerfile

```diff
@@ -18,41 +18,104 @@
 #    * Run unit tests:               docker run --rm -v WALinuxAgent:/home/waagent/WALinuxAgent python2.6 bash --login -c run-tests
 #    * Run tests that require root:  docker run --user root --rm -v WALinuxAgent:/home/waagent/WALinuxAgent python2.6 bash --login -c run-sudo-tests
 #
-FROM ubuntu:16.04
+FROM mcr.microsoft.com/mirror/docker/library/ubuntu:24.04
 ARG PYTHON_VERSION
 LABEL description="Test environment for WALinuxAgent"
 
 SHELL ["/bin/bash", "-c"]
 
-RUN \
-    apt-get update && \
-    apt-get -y install curl bzip2 sudo && \
-    groupadd waagent && \
-    useradd --shell /bin/bash --create-home -g waagent waagent && \
-    curl -sSf --retry 5 -o /tmp/python-${PYTHON_VERSION}.tar.bz2 https://dcrdata.blob.core.windows.net/python/python-${PYTHON_VERSION}.tar.bz2 && \
-    tar xjf /tmp/python-${PYTHON_VERSION}.tar.bz2 --directory / && \
-    rm -f /tmp/python-${PYTHON_VERSION}.tar.bz2 && \
-    echo $'\
-        \n\
-        cd /home/waagent \n\
-        source /home/waagent/virtualenv/python'${PYTHON_VERSION}/bin/activate$' \n\
-        function run-tests { \n\
-            nosetests --verbose --ignore-files test_cgroupconfigurator_sudo.py /home/waagent/WALinuxAgent/tests \n\
-        } \n\
-        function run-sudo-tests { \n\
-            nosetests --verbose /home/waagent/WALinuxAgent/tests/ga/test_cgroupconfigurator_sudo.py \n\
-        } \n\
-        ' | tee -a /home/waagent/.profile >> ~/.profile && \
-    sed -i 's/mesg n || true/tty -s \&\& mesg n/' ~/.profile && \
-    :
-
-#
-# TODO: Some unit tests create helper scripts that use 'python3' as shebang; we should probably port them to Bash, but installing Python 3 as a workaround for now.
-#
-RUN \
-    if [[ "${PYTHON_VERSION}" == "2.6" ]]; then \
-      apt-get -y install python3; \
+RUN <<..
+    #
+    # Install the Python venv
+    #
+    apt-get update
+    apt-get -y install curl bzip2 sudo
+    groupadd waagent
+    useradd --shell /bin/bash --create-home -g waagent waagent
+    curl -sSf --retry 5 -o /tmp/python-${PYTHON_VERSION}.tar.bz2 https://dcrdata.blob.core.windows.net/python/python-${PYTHON_VERSION}.tar.bz2
+    tar xjf /tmp/python-${PYTHON_VERSION}.tar.bz2 --directory /
+    chown -R waagent:waagent /home/waagent  # The UID:GID in the tarball may not match those of the user, so we need to fix that.
```

Comment:
Ubuntu 24 has a new user, ubuntu, with PID 1000 so now the PID in the tarball does not match waagent's